### PR TITLE
Issue15

### LINKED ISSUE #15: Compute histograms of inter-motif distances
Given a set of motifs and a set of sequences, compute histograms of inter-motif distances for all pairs of motifs and return pairs for which there is a significantly preferred distance. That is, loop across all sequences and record the distance between that start positions of the two motifs in every sequence. Note the symmetry, namely that distances between motif1 and motif2 will be the negatives of distances between motif2 and motif1. To not deal get twice the same underlying configuration, we can keep track only of non-negative distances. If there really is a preferred distance, this distance should occur more often than all others. We will have to decide on a cutoff. Here we could use the statistics we compute for generating the randomized sequences to calculate how often we expect specific motifs at specific distances. But maybe a simpler rule of thumb would work well enough, e.g. for each motif, the most frequent distance, as long as it's x fold higher than the average or than the second most common distance.

### PR DESCRIPTION
**Pull request by dcpb0**
### Description

Compute histograms of inter-motif distances.
Fixes # (15)

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- [ ] Documentation update

### Checklist

Please carefully read these items and tick them off if the statements are true
_or do not apply_.

- [x] I have performed a self-review of my own code
- [x] My code follows the existing coding style, lints and generates no new
      warnings
- [x] I have added type annotations to all function/method signatures, and I
      have added type annotations for any local variables that are non-trivial,
      potentially ambiguous or might otherwise benefit from explicit typing.
- [x] I have commented my code in hard-to-understand areas
- [ ] I have added ["Google-style docstrings"] to all new modules, classes,
      methods/functions or updated previously existing ones
- [ ] I have added tests that prove my fix is effective or that my feature
      works
- [ ] New and existing unit tests pass locally with my changes and I have not
      reduced the code coverage relative to the previous state
- [ ] I have updated any sections of the app's documentation that are affected
      by the proposed changes

If for some reason you are unable to tick off all boxes, please leave a
comment explaining the issue you are facing so that we can work on it
together.

Comment:
We have used the test files to see if the code is working. Also, we have documented in the code. We think it is clear what is doing what. However, we are not sure about the unticked boxes what they exactly mean.

### CODE REVIEW

File: issue15_code.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+def motifHistogram(seq1, seq2, directory,nrbins):
+    
+    #-------------------------------------------------------------------------
+    #create motif file of the 2 input motifs forward and revers
+    instances = [Seq(seq1), Seq(seq2)]
+    m = motifs.create(instances)
+    r = m.reverse_complement()
+    #-------------------------------------------------------------------------
+    
+    #get all reads of files in the folder
```
Comment (mzavolan) [inline issue15_code.py:28]: The script is doing more than specified, the input that was specified was a set of sequences.

File: issue15_code.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            motif2r=[]
+        
+            # saves the positions in the forward motifs
+            for pos, seq in m.instances.search(sequence): 
+                if seq == m.instances[0]: #saves the positions of the first motif
+                    motif1f.append(pos)
+                if seq == m.instances[1]: #saves the positions of the second motif
+                    motif2f.append(pos)
+                    
+            # saves the positions of the reverse motifs
```
Comment (mzavolan) [inline issue15_code.py:75]: Reverse complement is not needed.

File: issue15_code.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    motif2f.append(pos)
+                    
+            # saves the positions of the reverse motifs 
+            for pos, seq in r.instances.search(sequence): 
+                if seq == r.instances[0]: #saves the positions of the first motif
+                    motif1r.append(pos)
+                if seq == r.instances[1]: #saves the positions of the second motif
+                    motif2r.append(pos)
+     
+            # calculate  distance between forward motif1 and motif2, and save it in distance array
```
Comment (mzavolan) [inline issue15_code.py:82]: Positive and negative distances are not equivalent. The specification of the issue describes how they should be handled.

**Comment by uniqueg** [comment]
Not sure how good substance is (doing quite a bit of extra work, may include some faulty code), but left open as it still might be of use. Check carefully to make sure code adheres to spec defined in #15.

**Comment by uniqueg** [comment]
Closing this for now as we chose to implement a brute force algorithm for 3' adapter extraction for the meantime.

**Pull request closed without merging**